### ISSUE TITLE
VS Code Extension XAML/C# IntelliSense does not work with .NET 6.0

### ISSUE THREAD

**Issue by microhobby** [unoplatform/uno#8300]
### Current behavior

Using the latest version from MarketPlace I'm having the follow error when trying to edit the file `Shared\MainPage.xaml.cs`:

```
'MainPage' does not contain a definition for 'InitializeComponent' and no accessible extension method 'InitializeComponent' accepting a first argument of type 'MainPage' could be found (are you missing a using directive or an assembly reference?) [unonet6.Skia.Gtk]csharp(CS1061)
```

Also when editing the `Shared\MainPage.xaml` I'm having a lot of fails in the `Uno Platform - XAML` output panel:

```
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
```

<details>
<summary>Uno Platform - XAML logs</summary>

```
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: c:\Users\mpro3\source\unonet6
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: c:\Users\mpro3\source\unonet6
info: Uno.LSP.Roslyn.RoslynServerTarget[0]
      Compilation succeeded (_knownNamespaces:2, _typeNamespaceLookup:555)
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in 
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119
fail: Uno.LSP.Roslyn.RoslynServerTarget[0]
      RoslynServerTarget.Watcher.Failed: System.ArgumentException: The directory name 'obj' does not exist. (Parameter 'path')
         at System.IO.FileSystemWatcher.CheckPathValidity(String path)
         at System.IO.FileSystemWatcher..ctor(String path)
         at Uno.LSP.Roslyn.RoslynServerTarget.WatchChangesAsync(String projectPath, String configuration) in /home/vsts/work/1/s/src/server/Uno.LSP.Roslyn/RoslynServerTarget.Compilation.cs:line 119

```

</details>

### Expected behavior

intellissesense working, editing the xaml file working correctly and bringing symbols to the code behind without any issues.

### How to reproduce it (as minimally and precisely as possible)

I'm sharing the sample project.

- Open it on VS Code
- Open `Shared/MainPage.xaml.cs`
- Try to edit or add a new `Name` property to a component on the `Shared\MainPage.xaml`
- Try to access the `Name` created on the step above on the `Shared/MainPage.xaml.cs`

### Workaround

Using .NET 5 the first load of the project works as expected, you can access the symbols from the code behind on the `Shared/MainPage.xaml.cs`. But if you edit the `.xaml` the fails will occurs and will crash the intellissense.

### NuGet package version(s)

    <PackageReference Include="Uno.UI.Skia.Gtk" Version="4.1.8" />
    <PackageReference Include="Uno.UI.RemoteControl" Version="4.1.8" Condition="'$(Configuration)'=='Debug'" />

### Affected platforms

Skia.GTK

### IDE

VS Code

### IDE version

stable

### Relevant plugins

Uno 0.3
Omnisharp latest stable

**Comment by microhobby** [comment]
Opps, the sample project :
[unonet6.zip](https://github.com/unoplatform/uno/files/8227553/unonet6.zip)

**Comment by jeromelaban** [comment]
Would you know if this happens when the restore has been run already? /cc @ebariche

**Comment by microhobby** [comment]
> Would you know if this happens when the restore has been run already? /cc @ebariche

if I remember correctly, I tried with restore, build, reload the vs code window, but always with the same behavior ...

**Comment by ebariche** [comment]
Related to #8334

**Comment by jeromelaban** [comment]
I'm not able to reproduce this issue. If you can provide additional information on what exists in the file system when you're getting this error, it will be useful.

**Comment by jeromelaban** [comment]
Fixed in 0.4.0

**Comment by yongcong1415** [comment]
> Opps, the sample project : [unonet6.zip](https://github.com/unoplatform/uno/files/8227553/unonet6.zip)

I can run it OK..
at behind code can reference x:Name="nameInput",
